apps/fluent-tester/src/E2E/Avatar/pages/AvatarPageObject.ts

```diff
@@ -12,29 +12,29 @@ export const enum AvatarComponentSelector {
 }
 class AvatarPageObject extends BasePage {
   async getPrimaryComponentAttribute(attribute: string): Promise<string> {
-    return await this._primaryComponent.getAttribute(attribute);
+    return await (await this._primaryComponent).getAttribute(attribute);
   }
 
   async getSecondaryComponentAttribute(attribute: string): Promise<string> {
-    return await this._secondaryComponent.getAttribute(attribute);
+    return await (await this._secondaryComponent).getAttribute(attribute);
```

Comment:
Did not having this extra `await` cause issues? How was it okay before?

Reply:
We don't need to chain `await` commands, I don't believe this is necessary:

WDIO supports chaining commands without multiple `await`'s. See [this](https://blog.kiprosh.com/webdriverio-elements-chaining/)

Reply:
So I think we should remove all the chained awaits

Reply:
```
async function QueryWithChaining(identifier) {
  if (_rootView === null) {
    // Most of the elements we're searching for will be children of this _rootView node.
    _rootView = await $('~' + ROOT_VIEW);
  }
  const selector = '~' + identifier;
  let queryResult: WebdriverIO.Element;
  queryResult = await _rootView.$(selector);
  if (queryResult.error) {
    // In some cases, such as opened ContextualMenu items, the element nodes are not children of the _rootView node, meaning we need to start our search from the top of the tree.
    queryResult = await $(selector);
  }
  return queryResult;
  ```
  The problem here is that elements such as the ContextualMenu appear outside of the root, meaning that the chained query can fail. I have this check that returns a query from the top of the tree if the chained query fails, but this means that i have to return a promise rather than webdriver's chainablepromise.
  
  A workaround could be to have another query method specifically for elements using ContextualMenu.